storm-core/src/jvm/backtype/storm/scheduler/SupervisorDetails.java

```diff
@@ -119,10 +119,18 @@ private Double getTotalResource(String type) {
     }
 
     public Double getTotalMemory() {
-        return getTotalResource(Config.SUPERVISOR_MEMORY_CAPACITY_MB);
+        Double totalMemory = getTotalResource(Config.SUPERVISOR_MEMORY_CAPACITY_MB);
+        if (totalMemory == null) {
+            throw new IllegalStateException("default value for supervisor.memory.capacity.mb is not set!");
+        }
+        return totalMemory;
     }
 
     public Double getTotalCPU() {
-        return getTotalResource(Config.SUPERVISOR_CPU_CAPACITY);
+        Double totalCPU = getTotalResource(Config.SUPERVISOR_CPU_CAPACITY);
+        if (totalCPU == null) {
+            throw new IllegalStateException("default value for supervisor.cpu.capacity is not set!");
+        }
+        return totalCPU;
```

Comment:
I just meant something like:

```Java
throw new IllegalStateException("default value for "+Config.SUPERVISOR_CPU_CAPACITY+" is not set!");
```

However, as with the other logic that is checking for the mere presence of an entry in defaults.yaml, this should really be a unit test. I think it would be much better to move these kinds of checks into unit tests.  Moving to a unit test would simplify the code here by getting rid of unnecessary runtime checks while still helping developers find when defaults are missing.